### PR TITLE
Separate public interfaces

### PR DESCRIPTION
**Pull request by tock-ibm**
Signed-off-by: Yoav Tock <tock@il.ibm.com>

### CODE REVIEW

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/Iterator.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package org.hyperledger.orion.sdk;
+
+/**
+ * Iterator implements methods to iterate over a set records.
+ * 
+ */
+public interface Iterator {
```
Comment (liran-funaro) [inline orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/Iterator.java:10]: Java already has an Iterator interface

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/Query.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	 * the records.
+	 * 
+	 * @param dbName
+	 * @param startKey
+	 * @param endKey
+	 * @param limit
+	 * @return
+	 * @throws Exception
+	 */
+	Iterator GetDataByRange(String dbName, String startKey, String endKey, int limit) throws Exception;
```
Comment (liran-funaro) [inline orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/Query.java:73]: Methods begin with lowercase

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	 * 
+	 * @param endpoint
+	 * @param httpMethod
+	 * @param postData
+	 * @param signature
+	 * @return
+	 * @throws InterruptedException
+	 * @throws IOException
+	 */
+	HttpResponse<String> Query(String endpoint, String httpMethod, byte[] postData, byte[] signature)
```
Comment (liran-funaro) [inline orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java:33]: Methods begin with lowercase

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	 * transaction processing to complete and will return tx receipt as result. In
+	 * case of commitTimeout, http.StatusAccepted returned.
+	 * 
+	 * @param endpoint
+	 * @param msg
+	 * @return
+	 * @throws InterruptedException
+	 * @throws IOException
+	 */
+	HttpResponse<String> Submit(String endpoint, Message msg) throws InterruptedException, IOException;
```
Comment (liran-funaro) [inline orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java:52]: Methods begin with lowercase

**Review by liran-funaro — APPROVED** [review]

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/Iterator.java (hunk 1 above)
Comment (tock-ibm) [reply orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/Iterator.java:10]: We'll address this in the next release in order to not break the API for our i4Q users at this point. See #7

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java (hunk 3 above)
Comment (tock-ibm) [reply orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java:33]: We'll address this in the next release in order to not break the API for our i4Q users at this point. See #8

File: orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java (hunk 4 above)
Comment (tock-ibm) [reply orion-sdk-java/src/main/java/org/hyperledger/orion/sdk/RestClient.java:52]: We'll address this in the next release in order to not break the API for our i4Q users at this point. See #8

**Pull request merged**